src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration.Tests/ConfigurationLiveTests.cs

```diff
@@ -169,28 +165,142 @@ public async Task Set()
         }
 
         [Test]
-        public async Task Add()
+        public async Task SetKeyValue()
         {
             var connectionString = Environment.GetEnvironmentVariable("AZ_CONFIG_CONNECTION");
             Assert.NotNull(connectionString, "Set AZ_CONFIG_CONNECTION environment variable to the connection string");
             var service = new ConfigurationClient(connectionString);
 
+            string key = string.Concat("key-", Guid.NewGuid().ToString("N"));
+
             try
             {
-                Response<ConfigurationSetting> response = await service.AddAsync(s_testSetting);
+                string value = "my_value";
+                ConfigurationSetting setting = await service.SetAsync(key, value);
 
-                Assert.True(response.TryGetHeader("ETag", out string etagHeader));
+                Assert.AreEqual(key, setting.Key);
+                Assert.AreEqual(value, setting.Value);
+            }
+            finally
+            {
+                await service.DeleteAsync(key);
+            }
+        }
+
+        [Test]
+        public async Task AddExistingSetting()
```

Comment:
Should this test be called AddWithETag? This test throws the exception because ETag is present, not because the item is already present.

Reply:
Interesting... Short answer yes .
I was assuming that if the ETag was present, then that means that the setting is already in the configuration store, so why even ask the server.

Reply:
After talking to @xiangyan99 There isn't much reason to check or thorw exception in the presence of an etag, so I'm going to delete that check in the code and I will change this test to make sure it tests that we will get a 412 when the setting is already present.